version: '3'

networks:
  fake-account-api-test:
    driver: bridge

services:
  accountapitests:
    container_name: "account-api-tests"
    build:
      context: .
    depends_on:
      - accountapi
    networks:
      - fake-account-api-test
  accountapi:
    image: form3tech/interview-accountapi:v1.0.0-46-gb3b6c766
    container_name: "account-api"
    restart: on-failure
    depends_on:
      - postgresql
      - vault
    networks:
      - fake-account-api-test
    environment:
      - VAULT_ADDR=http://vault:8200
      - VAULT_TOKEN=8fb95528-57c6-422e-9722-d2147bcba8ed
      - PSQL_USER=root
      - PSQL_PASSWORD=password
      - PSQL_HOST=postgresql
      - PSQL_PORT=5432
      - STACK_NAME=f3-interview-accountapi
      - DATABASE-HOST=postgresql
      - DATABASE-SSL-MODE=disable
      - DATABASE-USERNAME=interview_accountapi_user
      - DATABASE-PASSWORD=123
    ports:
      - 8080:8080
  postgresql:
    image: postgres:9.5-alpine
    healthcheck:
      test: [ "CMD", "pg_isready", "-q", "-d", "postgres", "-U", "root" ]
      timeout: 45s
      interval: 10s
      retries: 10
    restart: always
    networks:
      - fake-account-api-test
    environment:
      - POSTGRES_USER=root
      - POSTGRES_PASSWORD=password
    volumes:
      - ./scripts/db:/docker-entrypoint-initdb.d/

  vault:
    image: vault:1.6.3
    networks:
      - fake-account-api-test
    environment:
      - SKIP_SETCAP=1
      - VAULT_DEV_ROOT_TOKEN_ID=8fb95528-57c6-422e-9722-d2147bcba8ed
